stl_to_obj/stl_to_tscn.py

```diff
@@ -0,0 +1,259 @@
+    sub_res_addon = ""
+    ext_res_addon = ""
+
+    # check if it's a PBR texture (require .jpgs)
+    if texture in Textures.texture_dict:
+        sub_res_addon += f'[sub_resource type="SpatialMaterial" id={sub_resource_id}]\n'
+        for jpg in Textures.texture_dict[texture].jpg_dict:
+            value = Textures.texture_dict[texture].jpg_dict[jpg]
+            # form the ext_resource
+            ext_res_addon += f'[ext_resource path="res://{Textures.TEXTURE_FOLDER}/{Textures.texture_dict[texture].folder}/{value}" type="Texture" id={ext_resource_id}]\n'
+            # add to sub_res_addon
+            if jpg in ["albedo", "roughness", "metallic"]:
+                sub_res_addon += f"{jpg}_texture = ExtResource( {ext_resource_id} )\n"
+            elif jpg == "normal":
+                sub_res_addon += Textures.NORMAL_OPTIONS + f"normal_texture = ExtResource( {ext_resource_id} )\n"
+            elif jpg == "depth":
+                sub_res_addon += Textures.DEPTH_OPTIONS + f"depth_texture = ExtResource( {ext_resource_id} )\n"
+
+            # don't forget to increment ext_resource_id for the next one
+            ext_resource_id += 1
+
+        # add uv1_scale if texture has it
+        if Textures.texture_dict[texture].uv1_scale:
+            sub_res_addon += f"uv1_scale = {Textures.texture_dict[texture].uv1_scale}\n"
+
+    elif texture in Textures.other_textures:
+        # only need sub_resource
+        sub_res_addon += f'[sub_resource type="SpatialMaterial" id={sub_resource_id}]\n{Textures.other_textures[texture]}'
+
+    return ext_res_addon, ext_resource_id, sub_res_addon 
+
+
+def main():
+    # argparse to take in input.txt
+    parser = argparse.ArgumentParser(prog = ".stl to .obj and .tscn file converter")
+    parser.add_argument("input_file", type = str, help = "input json filename")
+    args = parser.parse_args()
+
+    # read in input .json file
+    with open(args.input_file, 'r') as f:
+        ######################
+        # .tscn file strings #
+        ######################
+        # all the .obj, .glb, .jpg (textures)
+        ext_resource = ""
+        ext_resource_id = 1
+
+        # SpatialMaterials (construction of textures)
+        sub_resource = ""
+        sub_resource_id = 1
+
+        # MeshInstances and collision scenes (.glb)
+        nodes = ""
+
+
+        ###################
+        # parse json file #
+        ###################
+        # load JSON object as dictionary
+        data = json.load(f)
+
+        header = data["header"]
+
+        input_folder = header["input_folder"]
+
+        # make the output folder
+        output_folder = header["output_folder"]
+        try:
+            os.mkdir(f"./{output_folder}")
+        # delete the folder and remake it if it does exist
+        except OSError as error:
+            print(f"NOTICE: {output_folder} folder already exists")
+            shutil.rmtree(output_folder) # delete folder and contents
+            os.mkdir(f"./{output_folder}")
```

Comment:
put this in a function, and possibly add an argparse option to force this delete or not

Reply:
The delete always happens if the folder exists because we want to do a complete regeneration of the files. We don't want to write to a folder that exists and has files in it because then there could be extra files leftover that don't contribute to the .tscn file that is generated.